scootapi/setup/builder.go

```diff
@@ -0,0 +1,69 @@
+package setup
+
+import (
+	"fmt"
+	"os"
+	"path"
+)
+
+// Builder is able to create Scoot binaries
+type Builder interface {
+	// Scheduler returns the path to the Scheduler binary (or an error if it can't be built)
+	Scheduler() (string, error)
+	// Worker returns the path to the Worker binary (or an error if it can't be built)
+	Worker() (string, error)
+}
+
+const repoName = "github.com/scootdev/scoot"
+
+// GoBuilder uses the go command-line tools to build Scoot
+type GoBuilder struct {
+	cmds *Cmds
+
+	installed bool
+	err       error
+	schedBin  string
+	workerBin string
+}
+
+// NewGoBuilder creates a GoBuilder
+func NewGoBuilder(cmds *Cmds) *GoBuilder {
+	return &GoBuilder{
+		cmds: cmds,
+	}
+}
+
+// install runs go install and caches the results
+func (b *GoBuilder) install() {
+	if b.installed {
```

Comment:
It's fast enough not to matter, but it would be nice if this only compiled the minimum set of necessary binaries, or even just scheduler+worker.